peer/brontide.go

```diff
@@ -2538,13 +2544,59 @@ func (p *Brontide) reenableActiveChannels() {
 	// channel is already active, the update won't be sent.
 	for _, chanPoint := range activePublicChans {
 		err := p.cfg.ChanStatusMgr.RequestEnable(chanPoint, false)
-		if err == netann.ErrEnableManuallyDisabledChan {
-			p.log.Debugf("Channel(%v) was manually disabled, ignoring "+
-				"automatic enable request", chanPoint)
-		} else if err != nil {
-			p.log.Errorf("Unable to enable channel %v: %v",
-				chanPoint, err)
+
+		switch {
+		// No error occurred, continue to request the next channel.
+		case err == nil:
+			continue
+
+		// Cannot auto enable a manually disabled channel so we do
+		// nothing but proceed to the next channel.
+		case errors.Is(err, netann.ErrEnableManuallyDisabledChan):
+			p.log.Debugf("Channel(%v) was manually disabled, "+
+				"ignoring automatic enable request", chanPoint)
+
+			continue
+
+		// If the channel is reported as inactive, we will give it
+		// another chance. When handling the request, ChanStatusManager
+		// will check whether the link is active or not. One of the
+		// conditions is whether the link as been marked as
+		// reestablished, which happens inside a goroutine(htlcManager)
+		// after the link is started. And we may get a false negative
+		// saying the link is not active because that goroutine hasn't
+		// reached the line to mark the reestablishment. Thus we give
+		// it a second chance to send the request.
+		//
+		// TODO(yy): make the link signaling back its current state to
+		// properly handle this issue.
+		case errors.Is(err, netann.ErrEnableInactiveChan):
```

Comment:
the link calls `l.cfg.NotifyActiveChannel(*l.ChannelPoint())` when it's re-established which gives an ActiveChannelEvent via `SubscribeChannelEvents` but it's possible the notification will be missed. You could export `isReestablished` and make it a `ChannelLink` interface function and then check that

Reply:
I think you could:
- `SubscribeChannelEvents`
- call `RequestEnable`
- if `RequestEnable` fails, we know the link isn't active yet, so we continue listening for `ActiveChannelEvent`
- if `RequestEnable` succeeds, stop listening for the event and continue

Reply:
yeah I think the two systems are not on the same page. Somehow the `activePublicChans` here is not considered active by htlcswitch. 

> You could export isReestablished and make it a ChannelLink interface function and then check that

if the `htlcManager` hasn't marked it then this won't make a difference?

I think my question is, can any of the channels inside `activePublicChans` fail somehow? Maybe `RequestEnable` should block for some time if it sees the link is not reestablished...

Reply:
> I think my question is, can any of the channels inside activePublicChans fail somehow?

What do you mean here?

` Somehow the activePublicChans here is not considered active by htlcswitch.`

Because the reestablished atomic var isn't set?

I think the steps I wrote above should resolve this... maybe if `RequestEnable` fails for reasons other than inactive link, you retry, and otherwise wait for the active channel event (which is sent after the reestablish var in the link is set